app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -31,12 +35,55 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             return value;
         }
 
-        function handleFiltersUpdate(filters) {
+        function handleFiltersUpdate(filters, oldValue) {
+            /**
+             * TODO: to handle removal correctly we need to make sure we take oldValue into account,
```

Comment:
@rjmackay added a comment here while I handle this. Removal is a mess. Working on it.

Reply:
still thinking if this makes sense but I think it does.